### PR TITLE
log/include_correlation_id

### PR DESCRIPTION
**Pull request by aisha-deriv**

### CODE REVIEW

File: Log-Any-Adapter-DERIV-0.004/Changes [hunk 1]
```diff
@@ -0,0 +1,16 @@
+
```
Comment (chylli-deriv) [inline Log-Any-Adapter-DERIV-0.004/Changes:1]: this directory shouldn't be committed.

File: Changes [hunk 2]
```diff
@@ -2,6 +2,9 @@
 Revision history for Log-Any-Adapter-DERIV
 {{$NEXT}}
 
+0.004    2022-08-18 08:34:16+00:00 UTC
```
Comment (chylli-deriv) [inline Changes:5]: remove these 2 lines.
```suggestion
```

File: lib/Log/Any/Adapter/DERIV.pm [hunk 3]
```diff
@@ -359,6 +358,24 @@ sub log_entry {
     my $get_text =
         sub { my $color = shift // 0; $text_data{$color} //= $self->format_line($data, {color => $color}) . "\n"; return $text_data{$color}; };
 
+    # remove substitution context from message
+    if ($data->{message})
```
Comment (chylli-deriv) [inline lib/Log/Any/Adapter/DERIV.pm:362]: need some unit tests.

File: lib/Log/Any/Adapter/DERIV.pm [hunk 4]
```diff
@@ -359,6 +358,27 @@ sub log_entry {
...
+        severity  => $data->{severity},
+    );    
+    
+    if ($self->{context} && ref($self->{context}) eq 'HASH'){
+        my @keys = keys %{$self->{context}}; # Get the keys from the context hash
+ 
+        foreach my $key (@keys) {
+            $log_data{$key} = $self->{context}->{$key};
+        }
+        my $json_string = $JSON->encode(\%log_data);
```
Comment (felipe-deriv) [inline lib/Log/Any/Adapter/DERIV.pm:378]:
```suggestion
        my $json_string = $JSON->encode({ %log_data, $self->{context}->%* });
```

File: lib/Log/Any/Adapter/DERIV.pm (hunk 3 above)
Comment (felipe-deriv) [reply lib/Log/Any/Adapter/DERIV.pm:362]: could it also be a one line if ?
```suggestion
    $data->{message} =~ s/\".*// if $data->{message}
```

Comment (felipe-deriv) [reply lib/Log/Any/Adapter/DERIV.pm:362]: why does it remove everything from the first `"` occurence?

**Pull request merged**